core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/ModuleConfigUpdateExecutor.java

```diff
@@ -79,24 +86,59 @@ public void update(String moduleKey) {
   }
 
   private void fetchProjectQualityProfiles(String moduleKey, Set<String> qProfileKeys, ModuleConfiguration.Builder builder) {
+    SearchWsResponse qpResponse;
     try (InputStream contentStream = wsClient.get("/api/qualityprofiles/search.protobuf?projectKey=" + StringUtils.urlEncode(moduleKey)).contentStream()) {
-      SearchWsResponse qpResponse = QualityProfiles.SearchWsResponse.parseFrom(contentStream);
-      for (QualityProfile qp : qpResponse.getProfilesList()) {
-        String qpKey = qp.getKey();
-        if (!qProfileKeys.contains(qpKey)) {
-          throw new IllegalStateException(
-            "Module '" + moduleKey + "' is associated to quality profile '" + qpKey + "' that is not in storage. Server storage is probably outdated. Please update server.");
+      qpResponse = QualityProfiles.SearchWsResponse.parseFrom(contentStream);
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to load module quality profiles", e);
+    }
+    for (QualityProfile qp : qpResponse.getProfilesList()) {
+      String qpKey = qp.getKey();
+      if (!qProfileKeys.contains(qpKey)) {
+        throw new IllegalStateException(
+          "Module '" + moduleKey + "' is associated to quality profile '" + qpKey + "' that is not in storage. Server storage is probably outdated. Please update server.");
+      }
+      builder.getMutableQprofilePerLanguage().put(qp.getLanguage(), qp.getKey());
+    }
+
+  }
+
+  private void fetchProjectQualityProfilesBefore5dot2(String moduleKey, ModuleConfiguration.Builder builder) {
+    WsResponse response = wsClient.get("/batch/project?key=" + StringUtils.urlEncode(moduleKey));
+    try (JsonReader reader = new JsonReader(response.contentReader())) {
+      reader.beginObject();
+      while (reader.hasNext()) {
+        String propName = reader.nextName();
+        if ("qprofilesByLanguage".equals(propName)) {
+          reader.beginObject();
+          while (reader.hasNext()) {
+            String language = reader.nextName();
+            reader.beginObject();
+            while (reader.hasNext()) {
+              String qpPropName = reader.nextName();
+              switch (qpPropName) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1301)